From 3a82d7b1eeb79e332c9976af38130db79bf8b83c Mon Sep 17 00:00:00 2001
From: Seevalamuthu Mariappan <seevalam@codeaurora.org>
Date: Fri, 27 Nov 2020 13:19:02 +0530
Subject: [PATCH] ath11k: fix setting rx_decap_mode in wmi init config

In hw_ops ath11k_init_wmi_config_ipq8074, rx_decap_mode was set to
TARGET_DECAP_MODE_NATIVE_WIFI for all modes. Due to this, traffic
failed in raw mode with software encryption.

Fix this by assiging TARGET_DECAP_MODE_RAW to rx_decap_mode for raw mode.

Signed-off-by: Seevalamuthu Mariappan <seevalam@codeaurora.org>
---
 drivers/net/wireless/ath/ath11k/hw.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/ath/ath11k/hw.c b/drivers/net/wireless/ath/ath11k/hw.c
index 795f9c4..88cbe59 100644
--- a/drivers/net/wireless/ath/ath11k/hw.c
+++ b/drivers/net/wireless/ath/ath11k/hw.c
@@ -124,6 +124,9 @@ static void ath11k_init_wmi_config_ipq8074(struct ath11k_base *ab,
 	config->rx_timeout_pri[2] = TARGET_RX_TIMEOUT_LO_PRI;
 	config->rx_timeout_pri[3] = TARGET_RX_TIMEOUT_HI_PRI;
 	config->rx_decap_mode = TARGET_DECAP_MODE_NATIVE_WIFI;
+	if (test_bit(ATH11K_FLAG_RAW_MODE, &ab->dev_flags))
+		config->rx_decap_mode = TARGET_DECAP_MODE_RAW;
+
 	config->scan_max_pending_req = TARGET_SCAN_MAX_PENDING_REQS;
 	config->bmiss_offload_max_vdev = TARGET_BMISS_OFFLOAD_MAX_VDEV;
 	config->roam_offload_max_vdev = TARGET_ROAM_OFFLOAD_MAX_VDEV;
-- 
2.7.4

